crates/compiler/gen_llvm/src/llvm/build.rs

```diff
@@ -1165,7 +1165,12 @@ pub(crate) fn build_exp_expr<'a, 'ctx>(
             env.builder.position_at_end(check_if_null);
 
             env.builder.build_conditional_branch(
-                env.builder.build_is_null(tag_ptr, "is_tag_null"),
+                // have llvm optimizations clean this up
```

Comment:
nice. maybe we want to do it ourselves in the future given the latency of roc-pg earlier, but i hope llvm just entirely skips unreachable blocks

Reply:
with optimization it does for sure.